KP-Cratefiller/KPCF/fnc/fn_getItems.sqf

```diff
@@ -0,0 +1,76 @@
+/*
+    Killah Potatoes Cratefiller v1.1.0
+
+    Author: Dubjunk - https://github.com/KillahPotatoes
+    License: GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.html
+
+    Description:
+    Creates the item lists if generateLists is enabled.
+
+    Parameter(s):
+    NONE
+
+    Returns:
+    NONE
+*/
+
+// Reset variables
+KPCF_weapons = [];
+KPCF_grenades = [];
+KPCF_explosives = [];
+KPCF_items = [];
+KPCF_backpacks = [];
+
+private _configClasses = [];
+private _classNames = [];
+private _type = [];
+
+// Get all classnames from config
+{
+    _configClasses append (
+        "
+            _type = (configName _x) call BIS_fnc_itemType;
+            (getNumber (_x >> 'scope') == 2) &&
+            ((_type select 0) != '') &&
+            ((_type select 0) != 'VehicleWeapon')
+        " configClasses _x
+    );
+} forEach [(configFile >> "CfgMagazines"), (configFile >> "CfgWeapons"), (configFile >> "CfgVehicles"), (configFile >> "CfgGlasses")];
+
+// Convert to classname
+{
+    _classNames pushBack (configName _x)
+} forEach _configClasses;
+
+// Black & whitelisting
+_classNames = _classNames - KPCF_blacklistedItems;
+_classNames append KPCF_whitelistedItems;
```

Comment:
Shouldn't it be only one or another? 🤔 

If you get `_classnames` by crawling, shouldn't it already have all whitelistedItems registered?

Maybe something like (pseudo code)
If elements in whitelistedItems, then _classnames = whitelistedItems
else crawl the configs to get all available items and remove everything which is listed in blacklistedItems

Reply:
Whitelisting is implemented as some kind of security feature, thought about items that won't be recognized like they should.

Reply:
Ah, I see.
As far as I know you've described it like that also in the config file. 👍